src/api/assetDetails.ts

```diff
@@ -104,6 +103,7 @@ export const useAssetDetailsList = (
           filter.assetType.includes(asset.assetType),
         )
       },
+      enabled: !!isApiLoaded(api),
```

Comment:
I'd rather return the skeleton component when the api is not loaded. Example: StakingDashborad.tsx. It's more safe and better solution to show loaders. Because I think the key of the query will still be created

Reply:
@vkulinich-cl Problem here is that we're displaying multiple skeletons based on different conditions - if api is loading, if assets details are loading, ... . This results in not really smooth experience for user - loaders / skeletons are not not infinitely loading. 

By adding all query dependencies into this `enabled` prop we make sure it runs only when it has everything it needs while being in loading state until then. 

From tanstack doc: 

```
When enabled is false:

- If the query has cached data, then the query will be initialized in the status === 'success' or isSuccess state.
- If the query does not have cached data, then the query will start in the status === 'loading' and fetchStatus === 'idle' state.
- The query will not automatically fetch on mount.
```

To me it looks like this behaviour is exacly what we want. It also simplifies the usage of this hooks, because we don't have to keep in mind all dependencies and handle that in components.

Reply:
Documentation can be found here - https://tanstack.com/query/v4/docs/react/guides/disabling-queries. Take a look and let me know if I misunderstood the behaviour.

Reply:
What I wanted to say, that is make more sense to show skeletons when rpc is not working or rpc is loading instead of showing an empty page. This state can be also caused by slow internet connect etc. This is how it works on other pages.
<img width="1377" alt="Screenshot 2023-07-27 at 14 00 56" src="https://github.com/galacticcouncil/HydraDX-ui/assets/115090120/45f73302-efca-49c1-a304-7b9606553b86">

Reply:
we can disable the query if there in api. But anyway, we have to add skeletons for that state